File: 240106/짝수만 2로 나누기/divide-even-numbers-by-2.py
def modify(arr):
    for i in range(len(arr)):
        if arr[i] % 2 == 0:
            arr[i] //= 2


n = int(input())
arr = list(map(int, input().split()))

modify(arr)
for elem in arr:
    print(elem, end=" ")

File: 240202/짝수만 2로 나누기/divide-even-numbers-by-2.js
function divideEven(arr) {
    let result ="";
    for (let e of arr) {
        if (e % 2 == 0) {
            e /= 2;
        }
        result += e + " ";
    }
    console.log(result);
}

const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("\n");

let n = Number(input[0]);
let arr = input[1].split(" ").map(Number);

divideEven(arr);